linter.py

```diff
@@ -0,0 +1,39 @@
+import os
+import re
+from typing import Pattern, List
+
+search_note_pattern: Pattern = re.compile(r'^\*\*Note\*\*')
+search_warning_pattern: Pattern = re.compile(r'^\*\*Warning\*\*')
+correct_note_pattern: Pattern = re.compile(r'^> \*\*Note\*\*')
+correct_warning_pattern: Pattern = re.compile(r'^> \*\*Warning\*\*')
+
+def find_regex_in_file_content(
+    search_pattern: Pattern,
+    correct_pattern: Pattern,
+    filenames: List[str]
+) -> bool:
+    """Function that looks for a regex pattern in the given file contents."""
+
+    exit_code = 0
+    for filename in filenames:
+        with open(filename) as file:
+            lines = file.readlines()
+        for i, line in enumerate(lines):
+            if search_pattern.search(line):
+                print(f'{filename}:{i+1}: error: wrong notation "{search_pattern.pattern}"; expected "{correct_pattern.pattern}"')
+                exit_code = 1
+    return True if exit_code == 1 else False
+
+all_markdown_filenames: List[str] = [
+    filename for filename in os.listdir() if filename.endswith('.md')
+]
+found_note_pattern: bool = find_regex_in_file_content(
+    search_note_pattern, correct_note_pattern, all_markdown_filenames
+)
+found_warning_pattern: bool = find_regex_in_file_content(
+    search_warning_pattern, correct_warning_pattern, all_markdown_filenames
+)
+
+if found_note_pattern or found_warning_pattern:
+    print("Lint Check Failed!!")
+    exit(1)
```

Comment:
Please put these into a `main()` function that gets called under `if __name__ == '__main__'` so that this code doesn't run when the file is only imported

Reply:
Thanks for the explanation. Updated,